### ISSUE TITLE
Black Window when right clicking

### ISSUE THREAD

**Issue by PRouzeau** [openscad/openscad#3725]
[Windows] [Bug]
Since a few weeks, I experienced a bug  on Openscad snapshots. When right clicking on an image, as you do when searching the code associated with a model part (very nice feature, by the way), the list of codes open properly, but the underlying OpenScad windows became black, only the windows top bar remaining visible. The menu bar while invisible is still active and operating (meaning the menus open visibly and are active), the save is working, but it is impossible to recover the underlying OpenScad window. Docking/undocking have no effect. Trying to close/open a panel neither. The only way I found is to close/reopen OpenScad. As the save works, there is minimal damage (lost edit history) but personally as I right click accidentally quite frequently I met this bug a few times a day.
I am using Windows 64 bits version on up to date Windows 10 on a Dell Optiplex 3070. The graphic board is the Intel processor incorporated  UHD graphics 630 with drivers supplied by Dell.

![OpenScad_black_Window](https://user-images.githubusercontent.com/11719573/112136278-05feae00-8bcf-11eb-931f-347a4b8d137c.jpg)

**Comment by MichaelAtOz** [comment]
Similar screen issues in #3719.

**Comment by nophead** [comment]
I got this problem on my desktop PC when I was last at home but in my winter residence it seems to work properly on my Laptop. So perhaps it is graphics driver dependent.

**Comment by Parkinbot** [comment]
It ALSO may be driver dependent or graphics card dependent and most common drivers might catch this condition. My universal NVIDIA driver doesn't. 

However, since I don't get this effect with any other application on my machine (and I have installed many that also use OpenGL and hardware acceleration), I am 99.9% sure that there is a bug in OpenSCAD and that has to do with the popup window showing the module trace.

**Comment by dannypike** [comment]
I haven't examined all of the code to work out the correct solution, but I think that I've found a possible cause. This code in ```MouseSelector::select()``` in MouseSelector.cc:
```
  glDisable(GL_LIGHTING);
  glDepthFunc(GL_LESS);
  glCullFace(GL_BACK);
  glDisable(GL_CULL_FACE);
  glEnable(GL_DEPTH_TEST);
```
turns on the Z depth test. It never gets turned off again. That appears to be why the display disappears immediately afterwards - everything gets culled.

To prove this, I added a ```glDisable(GL_DEPTH_TEST)``` after it had extracted the color/index of the selected object and everything started behaving properly. As I said above, I'm not sure if this is the correct solution (or whether you want Z depth testing everywhere), but it should hopefully give you enough info to fix the problem.

**Comment by Parkinbot** [comment]
You made my day. 
It works for me on Windows 8.1, 64 Bit.

**Comment by t-paul** [comment]
Awesome, thanks @dannypike!

**Comment by dannypike** [comment]
You're welcome.

**Comment by OoDeLally** [comment]
I experience this as well on Ubuntu. What is the proper fix exactly? The link @t-paul mentioned does not work anymore.

**Comment by Matthew0x** [comment]
> I experience this as well on Ubuntu. I notice it goes away when unchecking the box `Enable for OpenGL 1.x` in `Preferences > Advanced`.

I have the same problem on Fedora and this setting doesn't fix it for me.
![image](https://github.com/user-attachments/assets/54c1b17c-63a9-4ea3-b4ce-9104e49abdeb)


When I right-click somewhere then the screen turns black.
Version 2021.01 from .RPM repository (the latest release on GitHub).

**Comment by nophead** [comment]
I have the problem with mice that have keybounce. The right click is interpreted as double right click, which is is show the code that generate the pixel. That gives a balck screen with some of my graphics cards.

**Comment by kintel** [comment]
This was fixed 3 years ago. Until we have a new stable release, use the latest development snapshot.

**Comment by Matthew0x** [comment]
I assumed so, but I had checked the tags, GH releases, system repository and the OpenSCAD website without seeing anything newer than 2021 and so I simply thought it was not available.

Upon your suggestion I scrolled back a bit further on the website and then I found it. The "Development Snapshots" seem available and even pre-compiled. I will try it out, thanks! Sorry for commenting on an old issue.

Too sweeten the inconvenience: I compared Flatpak (which I had been using) with .RPM and for some reason I get this bug when double right-clicking after hitting the preview button only. In Flatpak it appears to be working fine, although both packages report the same version. The others might be better off with just the Flatpak release.
